### PR TITLE
Add schemas, api endpoints w/functionality for queues/subscriptions

### PR DESCRIPTION
**Pull request by PrinceOm**

### CODE REVIEW

File: api/user/user.model.js [hunk 1]
```diff
@@ -3,11 +3,22 @@ const bcrypt = require('bcrypt-nodejs');
 const Promise = require('bluebird');
 
 const providers = ['local'];
+const oid = mongoose.Schema.Types.ObjectId;
+
+const queueSchema = new mongoose.Schema({
+  podcasts: [{ type: oid, required: true, ref: 'Podcast' }],
```
Comment (dting) [inline api/user/user.model.js]: Lets make this unique

File: api/user/user.model.js [hunk 2]
```diff
@@ -3,11 +3,22 @@ const bcrypt = require('bcrypt-nodejs');
...
+});
+
+const subscriptionSchema = new mongoose.Schema({
+  stations: [{ type: oid, required: true, ref: 'Station' }],
```
Comment (dting) [inline api/user/user.model.js]: Ditto

File: api/user/user.model.js (hunk 1 above)
Comment (PrinceOm) [inline api/user/user.model.js]: Done

File: api/user/user.model.js (hunk 2 above)
Comment (PrinceOm) [inline api/user/user.model.js]: Done

File: api/user/user.model.js [hunk 3]
```diff
@@ -3,11 +3,22 @@ const bcrypt = require('bcrypt-nodejs');
...
 
 const providers = ['local'];
+const oid = mongoose.Schema.Types.ObjectId;
+
+const queueItemSchema = new mongoose.Schema({
+  podcasts: { type: oid, required: true, ref: 'Podcast', unique: true },
+});
+
+const subscriptionSchema = new mongoose.Schema({
+  stations: { type: oid, required: true, unique: true },
```
Comment (dting) [inline api/user/user.model.js]: I think this needs to be to `ref: 'Station'`

**Comment by dting** [comment]
LGTM

**Pull request merged**